docs/create/aws/terraform/index-terraform.mdx

```diff
@@ -0,0 +1,597 @@
+
+```
+ terraform {
+ required_providers {
+  rediscloud = {
+    source = "RedisLabs/rediscloud"
+    version = "0.2.2"
+  }
+ }
+ }
+```
+
+
+
+
+The provider {} block configures the specific provider. In the below example, it is AWS.
+
+```
+ cloud_provider {
+ 
+   provider = "AWS"
+   cloud_account_id = 1
+   region {
+     region = "us-east-1"
+     networking_deployment_cidr = "10.0.0.0/24"
+     preferred_availability_zones = ["us-east-1a"]
+   }
+ }
+```
+
+
+
+### Resources
+
+Resources are the most important element in the Terraform language. This is where you describe the piece of infrastructure to be created, and this can range from a compute instance to defining specific permissions and much more.
+
+As shown below, the resource {} block is used to define components of your infrastructure. A resource might be a physical or virtual component, such as EC2, or it could be a logical component, such as a Heroku application.
+
+
+```
+ resource "random_password" "passwords" {
+ count = 2
+ length = 20
+ upper = true
+ lower = true
+ number = true
+}
+```
+
+
+Resource {} block has two strings before the block: resource types and resource names. The prefix of the type maps to the name of the provider. For example, the resource type “random_password” and the resource name “passwords” form a unique identifier of the resource. Terraform uses this ID to identify the resource.
+
+### Data sources
+
+Data sources allow Terraform to use information defined outside of Terraform, defined by another separate Terraform configuration, or modified by functions. Each provider may offer data sources alongside its set of resource types. A data source is accessed via a special kind of resource known as a data resource, declared using a data block.
+
+
+```
+ data "rediscloud_payment_method" "card" {
+ card_type = "Visa"
+ last_four_numbers = "XXXX"
+ }
+```
+
+
+
+A data block requests that Terraform read from a given data source ("rediscloud_payment_method") and export the result under the given local name ("card"). The name is used to refer to this resource from elsewhere in the same Terraform module, but has no significance outside of the scope of a module.
+Within the block body (between { and }) are query constraints defined by the data source. Most arguments in this section depend on the data source, and indeed in this example card_type and last_four_numbers are all arguments defined specifically for the rediscloud_payment_method data source.
+
+
+### Configure Redis Enterprise Cloud programmatic access
+
+In order to set up authentication with the Redis Enterprise Cloud provider, a programmatic API key must be generated for Redis Enterprise Cloud. The Redis Enterprise Cloud documentation contains the most up-to-date instructions for creating and managing your key(s) and IP access.
```

Comment:
Might be an idea to link to the relevant Redis Cloud docs here?

Reply:
Let me try to see if I can add link to Redis Enterprise cloud tutorial here.

Reply:
I have put the reference to our Redis Enterprise Cloud page. Have a look. @simonprickett